File: src/CompanionCore.Core/Conversations/Conversation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CompanionCore.Core.Conversations;

public class Conversation
{
    public Guid Id { get; set; }

    public Guid CompanionId { get; set; }

    public DateTime StartedAt { get; set; }

    public DateTime? LastMessageAt { get; set; }

    public string? Title { get; set; }

    public List<ConversationMessage> Messages { get; set; } = [];
}

File: src/CompanionCore.Web/src/models/Conversation.ts
export interface ConversationMessage {
    id: string;
    role: string;
    content: string;
    timestamp: string;
}

export interface Conversation {
    id: string;
    companionId: string;
    startedAt: string;
    lastMessageAt?: string;
    title?: string;
    messages: ConversationMessage[];
}